interpreter/interpreter_test.go

```diff
@@ -902,3 +902,32 @@ func TestStack(t *testing.T) {
 		t.Fatalf("unexpected stack -want/+got:\n%s", cmp.Diff(want, got))
 	}
 }
+
+func TestExistsNull(t *testing.T) {
+	node := semantic.File{
+		Body: []semantic.Statement{
+			&semantic.ExpressionStatement{
+				Expression: &semantic.UnaryExpression{
+					Operator: ast.ExistsOperator,
+					Argument: &semantic.IdentifierExpression{
+						Name: semantic.Symbol{
+							LocalName: "r",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	scope := values.NewScope()
+	scope.Set("r", values.Null)
+
+	intrp := interpreter.NewInterpreter(nil, nil)
+	sideEffects, _ := intrp.Eval(context.Background(), &node, scope, nil)
```

Comment:
This is great, and it also doesn't quite capture all of the places where we handle `exists`.

We actually evaluate expressions in two places, in the interpreter as you're doing here, and also in the `Compiler` which is a really confusing name. The `Compiler` is the thing that evaluates, e.g., the lambda that's passed into `map`. You could use an acceptance test to do this, if it doesn't produce an error.

Moreover, there's really two kinds of things that should get tested here:
- When we know the names and types of all the fields statically, e.g., what the input data to a function like `map` is produced with `array.from`
- When we don't know all the fields statically, e.g., when the input data is generated via `csv.from`

If I understand write #4626 would need to test both of these cases for both the interpreter and the compiler.